File: js/plazaNames.js
var nm1 = ["Accolade","Accomplishment","Achievement","Adventure","Ancestor","Animus","Anniversary","Apex","Art","Augury","Azure","Blue Moon","Capitol","Castle Square","Cathedral","Celebration","Central","Century","Ceremony","Cerulean","Champions'","Cherry Blossom","City Hall","Civic","Clover","Coconut","Commemoration","Community","Conquest","Crescent","Crystal","Daydream","Delicacy","Democracy","Desire","Devotion","Diamond","Discovery","Divinity","Dream","Dreamer","Duty","Effigy","Elegance","Elemental","Elysium","Emerald","Enchantment","Epiphany","Eternity","Euphony","Execution","Exploration","Fantasy","Festival","Firefly","Fortitude","Fortune","Founders'","Freedom","Galaxy","Gemstone","Glory","Golden","Good Omen","Grand","Grand Fountain","Grand Gala","Grand Harbor","Grand Market","Grand Meadow","Grand Palace","Grand Sanctum","Grand Splendor","Grand Treasure","Grapevine","Harbinger","Harmony","Heart's","Heaven's","Heroes'","History","Homage","Honor","Hope","Icewater","Illumination","Illusion","Independence","Infinity","Innocence","Innovation","Inspiration","Ivory","Jubilee","Justice","King","Kinship","Legion","Lemontree","Liberation","Liberty","Light","Lilypad","Love","Lovers'","Loyalty","Luminus","Lunar","Luxury","Magic","Majesty","Marble","Market","Marriage","Marvel","Melody","Memorial","Memory","Millennium","Miracle","Mirage","Monolith","Monument","Moonlight","Muse","Myriad","Mystic","Nature's","Nebula","New Haven","New Hope","Obelisk","Observation","Obsidian","Old Town","Opulence","Oracle","Orchard","Orchestra","Palm Leaf","Paradise","Paragon","Paramount","Passion","Peace Blossom","Peace Garden","Peacock","Pearl","People's","Perseverance","Phantasm","Phenomenon","Phoenix","Pilgrimage","Pinnacle","Pioneer","Pleasantview","Preservation","Prism","Prodigy","Promise","Prophecy","Prosperity","Public","Pyramid","Rainbow","Rebirth","Reflection","Refuge","Renaissance","Republic","Resonance","Revelation","Rose Petal","Royalty","Ruby","Sanctuary","Sanguine","Sapphire","Seduction","Sensation","Serenity","Settlers'","Silk","Silver","Silver Moon","Silver Oak","Silver Shrine","Sky","Snowflake","Solar","Songbird","Spring Gardens","Starlight","Statue","Summer Gardens","Supremacy","Sweet Delight","Sweetheart","Symphony","Temple","Tenacity","Timeless","Town Hall","Tranquility","Tribulation","Tribute","Trinity","Triumph","Trophy","Truth","Umbrella","Union","Unity","Universe","University","Utopia","Valiance","Velvet","Veneration","Venture","Vertex","Victory","Vigor","Vineyard","Virtue","Visionary","White Blossom","White Willow","Wishmaster","Wonder","Yesteryear","Zion"];
var nm2 = ["Plaza","Square"];
var br = "";

function nameGen(){
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd = Math.floor(Math.random() * nm1.length);
		rnd2 = Math.floor(Math.random() * nm2.length);
		names = nm1[rnd] + " " + nm2[rnd2];
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/plazaNames.py
__all__ = ['plazaNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm2', 'nameGen'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
            var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
            var.put('names', ((var.get('nm1').get(var.get('rnd'))+Js(' '))+var.get('nm2').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('Accolade'), Js('Accomplishment'), Js('Achievement'), Js('Adventure'), Js('Ancestor'), Js('Animus'), Js('Anniversary'), Js('Apex'), Js('Art'), Js('Augury'), Js('Azure'), Js('Blue Moon'), Js('Capitol'), Js('Castle Square'), Js('Cathedral'), Js('Celebration'), Js('Central'), Js('Century'), Js('Ceremony'), Js('Cerulean'), Js("Champions'"), Js('Cherry Blossom'), Js('City Hall'), Js('Civic'), Js('Clover'), Js('Coconut'), Js('Commemoration'), Js('Community'), Js('Conquest'), Js('Crescent'), Js('Crystal'), Js('Daydream'), Js('Delicacy'), Js('Democracy'), Js('Desire'), Js('Devotion'), Js('Diamond'), Js('Discovery'), Js('Divinity'), Js('Dream'), Js('Dreamer'), Js('Duty'), Js('Effigy'), Js('Elegance'), Js('Elemental'), Js('Elysium'), Js('Emerald'), Js('Enchantment'), Js('Epiphany'), Js('Eternity'), Js('Euphony'), Js('Execution'), Js('Exploration'), Js('Fantasy'), Js('Festival'), Js('Firefly'), Js('Fortitude'), Js('Fortune'), Js("Founders'"), Js('Freedom'), Js('Galaxy'), Js('Gemstone'), Js('Glory'), Js('Golden'), Js('Good Omen'), Js('Grand'), Js('Grand Fountain'), Js('Grand Gala'), Js('Grand Harbor'), Js('Grand Market'), Js('Grand Meadow'), Js('Grand Palace'), Js('Grand Sanctum'), Js('Grand Splendor'), Js('Grand Treasure'), Js('Grapevine'), Js('Harbinger'), Js('Harmony'), Js("Heart's"), Js("Heaven's"), Js("Heroes'"), Js('History'), Js('Homage'), Js('Honor'), Js('Hope'), Js('Icewater'), Js('Illumination'), Js('Illusion'), Js('Independence'), Js('Infinity'), Js('Innocence'), Js('Innovation'), Js('Inspiration'), Js('Ivory'), Js('Jubilee'), Js('Justice'), Js('King'), Js('Kinship'), Js('Legion'), Js('Lemontree'), Js('Liberation'), Js('Liberty'), Js('Light'), Js('Lilypad'), Js('Love'), Js("Lovers'"), Js('Loyalty'), Js('Luminus'), Js('Lunar'), Js('Luxury'), Js('Magic'), Js('Majesty'), Js('Marble'), Js('Market'), Js('Marriage'), Js('Marvel'), Js('Melody'), Js('Memorial'), Js('Memory'), Js('Millennium'), Js('Miracle'), Js('Mirage'), Js('Monolith'), Js('Monument'), Js('Moonlight'), Js('Muse'), Js('Myriad'), Js('Mystic'), Js("Nature's"), Js('Nebula'), Js('New Haven'), Js('New Hope'), Js('Obelisk'), Js('Observation'), Js('Obsidian'), Js('Old Town'), Js('Opulence'), Js('Oracle'), Js('Orchard'), Js('Orchestra'), Js('Palm Leaf'), Js('Paradise'), Js('Paragon'), Js('Paramount'), Js('Passion'), Js('Peace Blossom'), Js('Peace Garden'), Js('Peacock'), Js('Pearl'), Js("People's"), Js('Perseverance'), Js('Phantasm'), Js('Phenomenon'), Js('Phoenix'), Js('Pilgrimage'), Js('Pinnacle'), Js('Pioneer'), Js('Pleasantview'), Js('Preservation'), Js('Prism'), Js('Prodigy'), Js('Promise'), Js('Prophecy'), Js('Prosperity'), Js('Public'), Js('Pyramid'), Js('Rainbow'), Js('Rebirth'), Js('Reflection'), Js('Refuge'), Js('Renaissance'), Js('Republic'), Js('Resonance'), Js('Revelation'), Js('Rose Petal'), Js('Royalty'), Js('Ruby'), Js('Sanctuary'), Js('Sanguine'), Js('Sapphire'), Js('Seduction'), Js('Sensation'), Js('Serenity'), Js("Settlers'"), Js('Silk'), Js('Silver'), Js('Silver Moon'), Js('Silver Oak'), Js('Silver Shrine'), Js('Sky'), Js('Snowflake'), Js('Solar'), Js('Songbird'), Js('Spring Gardens'), Js('Starlight'), Js('Statue'), Js('Summer Gardens'), Js('Supremacy'), Js('Sweet Delight'), Js('Sweetheart'), Js('Symphony'), Js('Temple'), Js('Tenacity'), Js('Timeless'), Js('Town Hall'), Js('Tranquility'), Js('Tribulation'), Js('Tribute'), Js('Trinity'), Js('Triumph'), Js('Trophy'), Js('Truth'), Js('Umbrella'), Js('Union'), Js('Unity'), Js('Universe'), Js('University'), Js('Utopia'), Js('Valiance'), Js('Velvet'), Js('Veneration'), Js('Venture'), Js('Vertex'), Js('Victory'), Js('Vigor'), Js('Vineyard'), Js('Virtue'), Js('Visionary'), Js('White Blossom'), Js('White Willow'), Js('Wishmaster'), Js('Wonder'), Js('Yesteryear'), Js('Zion')]))
var.put('nm2', Js([Js('Plaza'), Js('Square')]))
pass
pass


# Add lib to the module scope
plazaNames = var.to_python()